improver_tests/spotdata/test_SpotExtraction.py

```diff
@@ -79,21 +76,60 @@ def setUp(self):
             iris.coords.CellMethod("maximum", coords="time", intervals="1 hour"),
         )
 
-        diagnostic_cube_xy = iris.cube.Cube(
-            diagnostic_data,
-            standard_name="air_temperature",
-            units="K",
-            dim_coords_and_dims=[(ycoord, 1), (xcoord, 0)],
-            attributes=attributes,
-            cell_methods=self.cell_methods,
-        )
-        diagnostic_cube_yx = iris.cube.Cube(
+        time = dt(2020, 6, 15, 12)
+        frt = time - timedelta(hours=6)
+
+        diagnostic_cube_yx = set_up_variable_cube(
             diagnostic_data.T,
-            standard_name="air_temperature",
+            name="air_temperature",
             units="K",
-            dim_coords_and_dims=[(ycoord, 0), (xcoord, 1)],
             attributes=attributes,
-            cell_methods=self.cell_methods,
+            domain_corner=(0, 0),
+            grid_spacing=10,
+            time=time,
+            frt=frt,
+        )
+        diagnostic_cube_yx.cell_methods = self.cell_methods
+
+        diagnostic_cube_xy = diagnostic_cube_yx.copy()
+        enforce_coordinate_ordering(
+            diagnostic_cube_xy,
+            [
+                diagnostic_cube_xy.coord(axis="x").name(),
+                diagnostic_cube_xy.coord(axis="y").name(),
+            ],
+            anchor_start=False,
+        )
+
+        times = np.array([time + timedelta(hours=i) for i in range(-3, 2)])
+        bounds = np.array(
+            [
+                [time - timedelta(hours=0.5), time + timedelta(hours=0.5)]
+                for time in times
+            ]
+        )
+        times = np.broadcast_to(times, (5, 5))
+        bounds = np.broadcast_to(bounds, (5, 5, 2))
+
+        self.time_aux_coord = iris.coords.AuxCoord(times, "time", bounds=bounds)
+        diagnostic_cube_2d_time = diagnostic_cube_yx.copy()
+        diagnostic_cube_2d_time.coord("time").rename("time_in_local_timezone")
+        diagnostic_cube_2d_time.add_aux_coord(self.time_aux_coord, data_dims=(0, 1))
+
+        self.expected_spot_time_coord = iris.coords.AuxCoord(
+            [
+                dt(2020, 6, 15, 9, 0),
+                dt(2020, 6, 15, 9, 0),
+                dt(2020, 6, 15, 11, 0),
+                dt(2020, 6, 15, 11, 0),
```

Comment:
I notice that there's a bit of inconsistency between the unit test here and the acceptance test regarding whether the time point is the midpoint between the bounds or is equal to the upper bound. As far as I was aware for all other quantities with bounds, we set to point equal to the upper bound. If we want to do something different here, then that will have implications for users.

Reply:
We don't Gavin, I just wrote stupid tests, and you are allowed to say so :-)

Modified to reflect normal placement of the points and bounds in the time coordinate.